wbooks/src/App/Home/Main/BookDetail/index.js

```diff
@@ -1,31 +1,61 @@
 import React from 'react';
 import { Link } from 'react-router-dom';
 
-import * as propTypes from '../../../constants/propTypes';
+import { booksPropTypes } from '../../../constants/propTypes';
+import BookService from '../../../services/books-service';
 
 import Details from './Details';
 import Recommendations from './Recommendations';
 import CommentsSection from './CommentsSection';
 
 import './styles.css';
 
-const books = require('../../../../books.json');
-
 class BookDetail extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: true,
+      error: false,
+      book: {}
+    };
+  }
+  componentDidMount() {
+    BookService.getBook(this.props.match.params.id).then(
+      book => {
+        this.setState({
+          loading: false,
+          book
+        });
+      },
+      () => {
+        this.setState({
+          loading: false,
+          error: true
+        });
+      }
+    );
+  }
+
   render() {
+    if (this.state.error) {
+      return <span>ERROR</span>;
+    }
+    if (this.state.loading) {
+      return <span>Cargando...</span>;
+    }
```

Comment:
You can extract this behavior to another component.
We can do it in anohter pr.